src/wp-includes/functions.php

```diff
@@ -824,7 +825,7 @@ function wp_extract_urls( $content ) {
 			. '(?:'
 				. '\([\w\d]+\)|'
 				. '(?:'
-					. "[^`!()\[\]{};:'\".,<>«»“”‘’\s]|"
+					. "[^`!()\[\]{}:'\".,<>«»“”‘’\s]|"
```

Comment:
By removing the semi-colon from the regex, there are no semi-colons to replace in the call to `str_replace()`. The semi-colon should stay in, so that after using `html_entity_decode`, the `str_replace()` will remove any errant semi-colons.